### DISCUSSION TITLE
Heroku integration. Is it real?

### DISCUSSION THREAD

**Discussion by Menzorg** [nhost/nhost#867] [General]
Have a good day. I founded your service very interesting. How about to create add-on for heroku?

https://elements.heroku.com/addons

It is platform to start apps using postgres as db for example, but heroku has no addons (microservice, creating with app) for gql. Is it real to create some integration with heroku, your addon may use pg from heroku instance and you can be only hasura provider! To my mind it can be powerful with minimal code and resources, so all heroku community could be potential your community. I developed some custom addons before, so it is some template, which could use your api to create free instances with your api, or instances of hasura only, if it is real.

**Comment by elitan** [comment]
Sounds interesting. If you want, feel free to create such an add-on.

Our internal focus is to provide Postgres ourselves (which we do).

If you have a Postgres database at some external provider, you can connect it to your Hasura instance at Nhost. This works with all Postgres providers such as Heroku, Digital Ocean, Supabase, Neon.tech, etc.

**Reply by Menzorg** [reply]
Has nhost some api, for creating free instancies?

**Reply by Menzorg** [reply]
I meen something like rest or http api.

**Reply by elitan** [reply]
We have a GraphQL API: https://nhost.run/v1/graphql

For auth, we're using [Hasura Auth](https://github.com/nhost/hasura-auth).

**Reply by Menzorg** [reply]
> GraphQL API

Ok, but has non-authorized user permission to create free instancies? Or could this api be used to create new user, and after it create free instance on created account? Basic idea is:
Somebody creates heroku app. On creating stage user can choose addons for his app, for example "Postgres" (this addon allready exists), and it meens, that heroku exec some script in addon, which creates pg-instance, and return db-url like "postres://.......". And user can use this env in app. So this is the way, i think how "hnost-hasura" addon should works. If user on creating app choose "hnost-hasura" addon, it should ask db-url to Postgres from user, access nhost api, create free instance and return url to GraphQL API, which can be used in heroku app.

**Reply by Menzorg** [reply]
Or if you think nhost should create own pg-instancies ok, why not. I just want to save nhost resources with using heroku pg-instance.

**Reply by elitan** [reply]
> has non-authorized user permission to create free instances?

No. Use has to be signed in to create a Nhost project (free instance).

> could this api be used to create new user, and after it create free instance on created account?

Yes. You can see the endpoints for how to create a user in the Hasura Auth link above.

---

If you want to work on this add-on, feel free to do so. I think it's an exciting idea. But we're not going to move our Postgres hosting to Heroku.

**Reply by Menzorg** [reply]
> Yes. You can see the endpoints for how to create a user in the Hasura Auth link above.

Wow, cool. Ok, our project need to be hosted on heroku, but as i said it has no hasura addos. I tried some tricks like creating frankenstien which start hasura instance inside heroku app, but there is a limits for resourcies. Thanks for answers, i will schedule the development of this addon.

**Reply by elitan** [reply]
If you still use Nhost and connect your Heroku database to your Hasura instance at Nhost:

![CleanShot 2022-08-01 at 08 58 03](https://user-images.githubusercontent.com/331818/182090627-c6a6337c-1282-459f-a881-841abb285b13.png)

**Comment by Menzorg** [comment]
For this experiment i need some API from nhost, like "create free hasura instance whith database_url = xxxx"

**Reply by Menzorg** [reply]
By the way, addons are a way to make money in heroku. They provide some basic functionality, but the advanced one costs money, in fact Heroku users will eventually be converted into your customers
